src/ucs/sys/sock.c

```diff
@@ -237,3 +237,106 @@ const char* ucs_sockaddr_str(const struct sockaddr *sock_addr,
 
     return str;
 }
+
+ucs_status_t ucs_sockaddr_addr_cmp(const struct sockaddr *sa1,
+                                   const struct sockaddr *sa2,
+                                   int *result_p)
+{
+    if (sa1->sa_family != sa2->sa_family) {
+        ucs_error("unable to compare socket addresses with "
```

Comment:
do we really need error messages? may output to warning & return ```not equal```?

Reply:
in general it is ok for me, but my concern was about ucs_error - isn't it too aggressive? what about ucs_debug or ucs_warn?

and one more thing: what about do not report error on comparing of valid addresses even for case if family are not equal? just return "not equal"

@brminich @yosefe ideas?

Reply:
yes they should be not equal

Reply:
done